packages/components/src/internal/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -787,9 +787,10 @@ class SampleTypeDesignerImpl extends React.PureComponent<Props & InjectedBaseDom
                         title={'Updating Sample Type with Unique ID field' + (numNewUniqueIdFields !== 1 ? 's' : '')}
                         onCancel={this.onUniqueIdCancel}
```

Comment:
what happens when user click Cancel while the action is in progress? Should cancel be disabled when submitting?

Reply:
Yeah, I considered that.  Currently, the Cancel button does not get disabled, so the effect is not actually cancelling but closing the modal. I think it makes sense to also disable the Cancel when submitting, but I'll check on usages.